lua/state_lua.cpp

```diff
@@ -223,6 +231,64 @@ void pushUnits(lua_State* L, const std::vector<torchcraft::Unit>& units) {
   }
 }
 
+void updateUserValueAndPush(
+    lua_State* L,
+    torchcraft::State* s,
+    const std::string& m,
+    int index) {
+  lua_getuservalue(L, index);
+
+  // Still up to date?
+  lua_getfield(L, -1, "__last_frame_from_bwapi");
+  if (lua_tointeger(L, -1) == s->frame_from_bwapi) {
```

Comment:
Could this break if you restart on the first frame (for example, when changing maps)?